### PR TITLE
#156577926 Allocate different stakeholders as owners of different HealthCheck alerts

### PR DESCRIPTION
**Pull request by kasigazibrian**
#### What does this PR do?
Allows allocation of different stakeholders as owners of different HealthChecks

#### Description of Task to be completed?
A user who has team access to another user's healthchecks account can view and get notified on the status of all the jobs defined in that account. This PR helps to reduce the team member healthchecks scope and allow stakeholder management by assigning the team member the relevant checks.

image

#### How should this be manually tested?
A user with team access creates several checks and a team.
He or she invites a member to join the team and selects the checks the team member should access.
A team member can then login and view only the checks he or she has been assigned

#### Any background context you want to provide?
Currently team access guarantees all existing individual healthchecks jobs access. A user granting team access to their account should be able to assign different jobs to the person they are inviting to join the team. 

#### What are the relevant pivotal tracker stories?

[#156577926](https://www.pivotaltracker.com/story/show/156577926)

### CODE REVIEW

**Comment by Rwothoromo** [comment]
@kasigazibrian This looks good to me.

**Comment by rogerokello** [comment]
Great work Brian.

**Comment by kasigazibrian** [comment]
@Rwothoromo thank you

**Comment by kasigazibrian** [comment]
@rogerokello thank you

File: hc/accounts/tests/test_profile.py [hunk 1]
```diff
@@ -69,18 +69,27 @@ def test_weekly_reports_data_posted(self):
...
 
+        form = {"invite_team_member": "1", "email": "bob@example.org"}
+        r = self.client.post("/accounts/profile/", form)
+        assert r.status_code == 200
+        member_emails = set()
+        for member in self.alice.profile.member_set.all():
+            member_emails.add(member.user.email)
+        self.assertContains(r, "is already a member of the team!")
+
+    def test_it_does_not_adds_team_member(self):
```
Comment (louicoder) [inline hc/accounts/tests/test_profile.py]: please correct the tense of this test function.

Comment (kasigazibrian) [reply hc/accounts/tests/test_profile.py]: @louiCoder Let me work on that

Comment (kasigazibrian) [reply hc/accounts/tests/test_profile.py]: I have implemented that @louiCoder

**Comment by louicoder** [comment]
good work @kasigazibrian . other than my comment, the rest is okay

File: hc/accounts/models.py [hunk 2]
```diff
@@ -81,18 +81,24 @@ def send_report(self):
...
 class Member(models.Model):
-    team = models.ForeignKey(Profile)
-    user = models.ForeignKey(User)
+    team = models.ForeignKey(Profile, on_delete=models.CASCADE)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    checks = models.ManyToManyField(Check)
+
+    def assign_checks(self, checks):
+        # checks = Check.objects.filter(user=self.user)
+        self.checks.add(*checks)
```
Comment (emugaya) [inline hc/accounts/models.py]: Add new line

File: hc/accounts/tests/test_switch_team.py [hunk 3]
```diff
@@ -26,5 +28,5 @@ def test_it_switches_to_own_team(self):
         self.client.login(username="alice@example.org", password="password")
 
         url = "/accounts/switch_team/%s/" % self.alice.username
-        r = self.client.get(url, follow=True)        
+        r = self.client.get(url, follow=True)
         self.assertEqual(r.status_code, 200)
```
Comment (emugaya) [inline hc/accounts/tests/test_switch_team.py]: Add new line

File: hc/accounts/models.py (hunk 2 above)
Comment (kasigazibrian) [reply hc/accounts/models.py]: @emugaya let me work on that

File: hc/accounts/tests/test_switch_team.py (hunk 3 above)
Comment (kasigazibrian) [reply hc/accounts/tests/test_switch_team.py]: @emugaya Let me work on that

File: hc/accounts/models.py (hunk 2 above)
Comment (kasigazibrian) [reply hc/accounts/models.py]: @emugaya I have implemented the changes

File: hc/accounts/tests/test_switch_team.py (hunk 3 above)
Comment (kasigazibrian) [reply hc/accounts/tests/test_switch_team.py]: @emugaya I have worked on that

**Review by emugaya — APPROVED** [review]
LGTM

**Pull request merged**